docs/README.md

```diff
@@ -0,0 +1,114 @@
+
+*Additional context*:
+
+Currently for EigenDA, the max operator count is 200. This maximum exists because EigenDA requires that completed "jobs" validate a signature by the aggregate BLS pubkey of the operator set over some job parameters. Although an aggregate BLS pubkey's signature should have a fixed cost no matter the number of operators, it may be the case that not all operators sign off on a job.
+
+When this happens, EigenDA needs to provide a list of the pubkeys of the non-signers to subtract them out from the quorum's aggregate pubkey ("Apk"). The limit of 200 operators keeps the gas costs reasonable in a worst case scenario. See `BLSSignatureChecker.checkSignatures` for this part of the implementation.
+
+In order to prevent the operator set from getting calcified, the churn mechanism was introduced to allow operators to be replaced in some cases. Future work is being done to increase the max operator count and refine the churn mechanism.
+
+##### Hooking Into EigenLayer Core
+
+The main thing that links an AVS to the EigenLayer core contracts is that when EigenLayer operators register/deregister with an AVS, the AVS calls these functions in EigenLayer core:
+* [`DelegationManager.registerOperatorToAVS`][core-registerToAVS]
+* [`DelegationManager.deregisterOperatorFromAVS`][core-deregisterFromAVS]
+
+These methods ensure that the operator registering with the AVS is also registered as an operator in EigenLayer core. In this repo, these methods are called by the `RegistryCoordinator`.
+
+Eventually, operator slashing and payment for services will be part of the middleware/core relationship, but these features aren't implemented yet and their design is a work in progress.
+
+### System Components
+
+#### Registries
+
+| Code | Type | Proxy |
+| -------- | -------- | -------- |
+| [`RegistryCoordinator.sol`](../src/RegistryCoordinator.sol) | Singleton | Transparent proxy |
+| [`BLSApkRegistry.sol`](../src/BLSApkRegistry.sol) | Singleton | Transparent proxy |
+| [`StakeRegistry.sol`](../src/StakeRegistry.sol) | Singleton | Transparent proxy |
+| [`IndexRegistry.sol`](../src/IndexRegistry.sol) | Singleton | Transparent proxy |
+
+The `RegistryCoordinator` is the primary entry point for operators as they register for and deregister from an AVS's quorums. When operators register or deregister, the registry coordinator updates that operator's currently-registered quorums, and pushes the registration/deregistration to each of the three registries it controls:
+* `BLSApkRegistry`: tracks the aggregate BLS pubkey hash for the operators registered to each quorum. Also maintains a history of these aggregate pubkey hashes.
+* `StakeRegistry`: interfaces with the EigenLayer core contracts to track historical state of operators per quorum.
```

Comment:
this is nice and short and sweet. perhaps adding that this is where operator weights are evaluated and stored?
"historical state" seems rather vague, and what this is focusing on is "stake" -- but that is clearly indicated by the name, so... idk. not strongly opinionated, just a suggestion

Reply:
[ac38452](https://github.com/Layr-Labs/eigenlayer-middleware/pull/96/commits/ac3845212bca0232f3db26882c12568b4a6e3660)